File: examples/node/asticaVoice_sample.js
var axios = require('axios');
var fs = require('fs');

var asticaAPI_key = 'YOUR API KEY'; // visit https://astica.ai
var asticaAPI_timeout = 10; // in seconds.

var asticaAPI_endpoint = 'https://voice.astica.ai/speak';
var asticaAPI_modelVersion = '1.0_full';

var asticaAPI_voiceid = 0; // see list of voice id: https://astica.ai/voice/documentation/
var asticaAPI_input = 'hello, how are you doing today?'; // text to be spoken
var asticaAPI_lang = 'en-US'; // language code

var asticaAPI_outputFile = 'output.wav' //save audio file of speech

// Define payload dictionary
var asticaAPI_payload = {
  tkn: asticaAPI_key,
  modelVersion: asticaAPI_modelVersion,
  input: asticaAPI_input,
  voice: asticaAPI_voiceid,
  lang: asticaAPI_lang,
};

axios.post(asticaAPI_endpoint, asticaAPI_payload , {
  headers: {
    'Content-Type': 'application/json'
  },
  timeout: asticaAPI_timeout * 1000,
}).then(function (response) {
  if (response.status == 200) {
    
    console.log('astica API Output:');
    console.log(JSON.stringify(response.data, null, 2));
    console.log('=================');
    
    //Handle asticaAPI response
    if (response.data.status == "error") {
      console.log('Output:\n',response.data.error);
    } else if (response.data.status == "success") {
      console.log("Success");
      
      var wavBuffer = new Buffer.from(response.data.wavBuffer.data);
      
      // Write the file on disk
      fs.writeFile(asticaAPI_outputFile, wavBuffer, function(err) {
        if (err) console.log('Error while saving file.');
      });
      
    } else {
      console.log(response.data);
    }
  } else {
    console.log('Failed to connect to the API.');
  }
}).catch(function (error) {
  console.log('Failed to connect to the API. Error: ', error.message);
});

File: examples/python/asticaVoice_sample.py
import requests
import json
import base64
import os


# API configurations
asticaAPI_key = 'YOUR API KEY'  # visit https://astica.ai
asticaAPI_timeout = 10 # in seconds.
asticaAPI_endpoint = 'https://voice.astica.ai/speak'
asticaAPI_modelVersion = '1.0_full'

asticaAPI_voiceid = 0; # see list of voice id: https://astica.ai/voice/documentation/
asticaAPI_input = 'hello, how are you doing today?' # text to be spoken
asticaAPI_lang = 'en-US' # language code

asticaAPI_outputFile = 'output.wav' #save audio file of speech
asticaAPI_outputPlayback = False #pip install sounddevice numpy
# Define payload dictionary
asticaAPI_payload = {
    'tkn': asticaAPI_key,
    'modelVersion': asticaAPI_modelVersion,
    'input': asticaAPI_input,
    'voice': asticaAPI_voiceid,
    'lang': asticaAPI_lang,
}



def asticaAPI(endpoint, payload, timeout):
    response = requests.post(endpoint, data=json.dumps(payload), timeout=timeout, headers={ 'Content-Type': 'application/json', })
    if response.status_code == 200:
        return response.json()
    else:
        return {'status': 'error', 'error': 'Failed to connect to the API.'}



# call API function and store result
asticaAPI_result = asticaAPI(asticaAPI_endpoint, asticaAPI_payload, asticaAPI_timeout)

# print API output
print('\nastica API Output:')
print(json.dumps(asticaAPI_result, indent=4))
print('=================')
# Handle asticaAPI response
if 'status' in asticaAPI_result:
    # Output Error if exists
    if asticaAPI_result['status'] == 'error':
        print('Output:\n', asticaAPI_result['error'])
    # Output Success if exists
    if asticaAPI_result['status'] == 'success':
        print("Success")
        
        #handle wav buffer
        wavData = bytes(asticaAPI_result['wavBuffer']['data'])
        
        #save wav file
        with open(asticaAPI_outputFile, 'wb') as f:
            f.write(wavData)
            
        if asticaAPI_outputPlayback: 
            #or play wav file
            import sounddevice as sd #pip install sounddevice
            import numpy as np
            wav_array = np.frombuffer(wavData, dtype=np.int16)
            # assuming audio data array contains 44100 samples per second
            fs = 16000  
            # play sound
            sd.play(wav_array, fs)
            # use this during sound playback, otherwise the sound will stop immediately
            sd.wait()    
        
else:
    print('Invalid response')
